File: SpringCloud_Study/e-commerce/users-service/src/main/java/com/example/usersservice/repository/UserRepository.java
package com.example.usersservice.repository;

import com.example.usersservice.entity.UserEntity;
import org.springframework.data.jpa.repository.JpaRepository;

// CrudRepository를 상속받아도 됨
public interface UserRepository extends JpaRepository<UserEntity, Long> {
    
}

File: Kotlin_Study/kotlin-security/src/main/kotlin/com/example/kotlinsecurity/repository/UserRepository.kt
package com.example.kotlinsecurity.repository

import com.example.kotlinsecurity.domain.User
import org.springframework.data.jpa.repository.JpaRepository

interface UserRepository : JpaRepository<User, Long> {
    fun findByName(name: String): User?
    fun findByPhone(phone: String): User?
}
